sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1379,6 +1392,220 @@ def test_warm_start_converge_LR():
+    # with saga solver (l1_ratio different from 0 or 1)
+    X, y = make_classification(random_state=0)
+
+    C = 1 / .5
+    l1_ratio = .5
+    coeffs = list()
+    for penalty in ('elastic-net', 'l1', 'l2'):
+        lr = LogisticRegression(penalty=penalty, C=C, solver='saga',
+                                random_state=0, l1_ratio=l1_ratio)
+        lr.fit(X, y)
+        coeffs.append(lr.coef_)
+
+    elastic_net_coeffs, l1_coeffs, l2_coeffs = coeffs
+    # make sure coeffs differ by at least .1
+    assert not np.allclose(elastic_net_coeffs, l1_coeffs, rtol=0, atol=.1)
+    assert not np.allclose(elastic_net_coeffs, l2_coeffs, rtol=0, atol=.1)
+    assert not np.allclose(l2_coeffs, l1_coeffs, rtol=0, atol=.1)
+
+
+@pytest.mark.parametrize('C', [.001, .1, 1, 10, 100, 1000, 10000000])
+def test_elastic_net_l1_l2_equivalence(C):
+    # Make sure elastic-net is equivalent to l1 when l1_ratio=1 and l2 when
+    # l1_ratio=0.
+    X, y = make_classification(random_state=0)
+
+    lr = LogisticRegression(penalty='elastic-net', C=C, l1_ratio=0,
+                            solver='saga', random_state=0)
+    lr.fit(X, y)
+    elastic_net_l1_zero_coeffs = lr.coef_
+
+    lr = LogisticRegression(penalty='l2', C=C, solver='saga', random_state=0)
+    lr.fit(X, y)
+    l2_coeffs = lr.coef_
+
+    assert_array_almost_equal(elastic_net_l1_zero_coeffs, l2_coeffs)
+
+    lr = LogisticRegression(penalty='elastic-net', C=C, l1_ratio=1,
+                            solver='saga', random_state=0)
+    lr.fit(X, y)
+    elastic_net_l1_one_coeffs = lr.coef_
+
+    lr = LogisticRegression(penalty='l1', C=C, solver='saga', random_state=0)
+    lr.fit(X, y)
+    l1_coeffs = lr.coef_
+
+    assert_array_almost_equal(elastic_net_l1_one_coeffs, l1_coeffs)
+
+
+@pytest.mark.parametrize('C', [.001, 1, 100, 10000000])
+def test_elastic_net_vs_l1_l2(C):
+    # Make sure that elastic net with grid search on l1_ratio gives same or
+    # better results than just l1 or just l2.
+
+    X, y = make_classification(500, random_state=0)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+    cv = StratifiedKFold(5, random_state=0)
+    param_grid = {'l1_ratio': np.linspace(0, 1, 10)}
+
+    enet_clf = LogisticRegression(penalty='elastic-net', C=C, solver='saga',
+                                  random_state=0)
+    gs = GridSearchCV(enet_clf, param_grid, cv=cv, iid=False, refit=True)
+
+    l1_clf = LogisticRegression(penalty='l1', C=C, solver='saga',
+                                random_state=0)
+    l2_clf = LogisticRegression(penalty='l2', C=C, solver='saga',
+                                random_state=0)
+
+    for clf in (gs, l1_clf, l2_clf):
+        clf.fit(X_train, y_train)
+
+    assert gs.score(X_test, y_test) >= l1_clf.score(X_test, y_test)
+    assert gs.score(X_test, y_test) >= l2_clf.score(X_test, y_test)
+
+
+def test_elastic_net_CV_grid_search():
+    # make sure ElasticNetCV gives same best params as GridSearchCV
```

Comment:
LogisticRegressionCV with elastic net? There's also ElasticNetCV which is regression